extensions/wikia/Email/EmailController.class.php

```diff
@@ -481,17 +481,34 @@ protected function getAvatarURL( \User $user ) {
 	}
 
 	protected function findUserFromRequest( $paramName, \User $default = null ) {
-		$userName = $this->getRequest()->getVal( $paramName );
-		if ( empty( $userName ) ) {
+		$user = $this->getRequest()->getVal( $paramName );
+		if ( empty( $user ) ) {
 			return $default;
 		}
 
 		// Allow an anonymous user to be specified
-		if ( $userName == self::ANONYMOUS_USER_ID ) {
+		if ( $user == self::ANONYMOUS_USER_ID ) {
 			return \User::newFromId( 0 );
 		}
 
-		return $this->getUserFromName( $userName );
+		if ( is_numeric( $user ) ) {
+			return $this->getUserFromId( $user );
+		}
+
+		return $this->getUserFromName( $user );
+	}
+
+	/**
+	 * Returns a user object from an id
+	 *
+	 * @param $userId
+	 *
+	 * @return \User
+	 * @throws Fatal
+	 * @throws \MWException
+	 */
+	private function getUserFromId( $userId ) {
```

Comment:
What's the purpose of this function? Why not just call `\User::newFromId( $userId )`?